### PR TITLE
Update and simplify some tests; Correct attrs testing and add info to docs

### PR DESCRIPTION
**Pull request by marcharper**
Simplifies some tests using the new `attrs` option in `TestPlayer.test_responses`.

### CODE REVIEW

File: docs/tutorials/contributing/strategy/writing_test_for_the_new_strategy.rst [hunk 1]
```diff
@@ -110,6 +110,14 @@ All three of these functions can take an optional keyword argument
    expected responses are D, C, C, C. Note that the histories will elongate as
    the responses accumulated.
 
+    The function :code:`responses_test` also accepts a dictionary parameter of
```
Comment (drvinceknight) [inline docs/tutorials/contributing/strategy/writing_test_for_the_new_strategy.rst]: Hehe: all the way through reading this I was like 'I hope he's documented this extra part' :) :+1:

**Comment by drvinceknight** [comment]
Can anything be done about the coveralls?

**Comment by marcharper** [comment]
As far as I can tell, there are not any fewer lines covered, they are just not covered as often.

**Comment by marcharper** [comment]
If you look at the coverage report, it's just fewer repetitions of some lines. No files had a decrease in coverage % -- coverage just counts lines executed and how many times, it doesn't know if tests are redundant. I'm not sure why the number is decreasing, but obviously it's a very small decrease.

**Comment by drvinceknight** [comment]
> If you look at the coverage report, it's just fewer repetitions of some lines. No files had a decrease in coverage % -- coverage just counts lines executed and how many times, it doesn't know if tests are redundant. I'm not sure why the number is decreasing, but obviously it's a very small decrease.

Sorry, after your other comment I checked and indeed meant to give this the all clear from me: someone came by etc...

:+1:

**Comment by marcharper** [comment]
@meatballs @langner this one is ready for your review/merge. no rush, just in case the  :x: is keeping you away.

**Pull request merged**